config/config.go

```diff
@@ -206,13 +292,19 @@ func (cfg Config) ValidReferer(reqURL, ref string) bool {
 		// just ensure ref is same host/scheme as req
 		u.Path = ""
 		u.RawQuery = ""
-		return strings.HasPrefix(ref, u.String())
+		matched, err := MatchURL(u.String(), ref)
+		if err != nil {
+			return false
+		}
+		return matched
 	}
 
 	for _, u := range cfg.Auth.RefererURLs {
-		if strings.HasPrefix(ref, u) {
-			return true
+		matched, err := MatchURL(u, ref)
+		if err != nil {
+			return false
 		}
+		return matched
```

Comment:
Same as above, since we're ranging over a list, only return true if `matched` is true